plugin_tests/featureset_test.py

```diff
@@ -0,0 +1,266 @@
+                }
+            ]
+        }
+
+        # Try to create a featureset as anonymous
+        resp = self.request(
+            path='/featureset', method='POST', params=basicFeaturesetParams)
+        self.assertStatus(resp, 401)
+
+        # Try to create a featureset without privileges
+        resp = self.request(
+            path='/featureset', method='POST', user=basicUser,
+            params=basicFeaturesetParams)
+        self.assertStatus(resp, 403)
+
+        # Try to create a featureset with an empty name
+        tempFeaturesetParams = basicFeaturesetParams.copy()
+        tempFeaturesetParams['name'] = ''
+        resp = self.request(
+            path='/featureset', method='POST', user=studyAdminUser,
+            type='application/json', body=json.dumps(tempFeaturesetParams))
+        self.assertStatus(resp, 400)
+
+        # Try to create a featureset with an invalid version
+        tempFeaturesetParams = basicFeaturesetParams.copy()
+        tempFeaturesetParams['version'] = 'foo'
+        resp = self.request(
+            path='/featureset', method='POST', user=studyAdminUser,
+            type='application/json', body=json.dumps(tempFeaturesetParams))
+        self.assertStatus(resp, 400)
+
+        # TODO: Try to create a featureset with invalid features
+
+        # List all (nonexistent) featuresets
+        resp = self.request(
+            path='/featureset', method='GET')
+        self.assertStatusOk(resp)
+        self.assertEqual(resp.json, [])
+
+        # Create a valid featureset
+        resp = self.request(
+            path='/featureset', method='POST', user=studyAdminUser,
+            type='application/json', body=json.dumps(basicFeaturesetParams))
+        self.assertStatusOk(resp)
+        # Lookup the database entry, for access to non-deterministic details
+        # like '_id' and 'created'
+        basicFeatureset = Featureset.load(resp.json['_id'])
+
+        # Check the list of all featuresets
+        resp = self.request(
+            path='/featureset', method='GET')
+        self.assertStatusOk(resp)
+        self.assertIsInstance(resp.json, list)
+        self.assertEqual(len(resp.json), 1)
+        self.assertDictEqual({
+            '_id': str(basicFeatureset['_id']),
+            'name': basicFeaturesetParams['name'].strip(),
+            'version': float(basicFeaturesetParams['version']),
+        }, resp.json[0])
+
+        # Check featureset details as anonymous
+        resp = self.request(
+            path='/featureset/%s' % basicFeatureset['_id'], method='GET')
+        self.assertStatusOk(resp)
+        self.assertDictEqual({
+            '_id': str(basicFeatureset['_id']),
+            '_modelType': 'featureset',
+            'name': basicFeaturesetParams['name'].strip(),
+            'version': float(basicFeaturesetParams['version']),
+            'created':
+                basicFeatureset['created'].replace(tzinfo=pytz.UTC).isoformat(),
```

Comment:
It could be cleaner to apply girder.utility.parseTimestamp` to the `created` field before comparison.